src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,160 @@
+                            statements: followingStatements
+                            );
+
+                        BoundUsingStatement boundUsing = new BoundUsingStatement(
+                            syntax: localDeclaration.Syntax,
+                            locals: ImmutableArray.Create<LocalSymbol>(),
+                            declarationsOpt: new BoundMultipleLocalDeclarations(
+                                localDeclaration.Syntax,
+                                localDeclarations),
+                            expressionOpt: null,
+                            iDisposableConversion: Conversion.Identity,
+                            disposeMethodOpt: null,
+                            body: innerBlock
+                            );
+                        precedingStatements.Add(boundUsing);
+
+                        BoundBlock outermostBlock = new BoundBlock(
+                            syntax: localDeclaration.Syntax,
+                            locals: node.Locals,
+                            statements: precedingStatements.ToImmutableArray<BoundStatement>());
+
+                        return outermostBlock;
+
+                    }
+                }
+                else if (statements[i] is BoundMultipleLocalDeclarations boundMultiple)
+                {
+                    if (boundMultiple.LocalDeclarations.Any())
+                    {
+                        if (boundMultiple.LocalDeclarations[0].LocalSymbol.IsUsing)
+                        {
+                            List<BoundStatement> precedingStatements = new List<BoundStatement>();
+                            for (int j = 0; j < i; j++)
+                            {
+                                precedingStatements.Add(statements[j]);
+                            }
+                            List<BoundStatement> followingStatements = new List<BoundStatement>();
+                            for (int k = i + 1; i < statements.Length; k++)
+                                followingStatements.Add(statements[k]);
+                            return LowerBoundMultipleLocalDeclarationUsingVar(boundMultiple, node.Locals, precedingStatements, followingStatements.ToImmutableArray());
+                        }
+                    }
+                }
+            }
+            return node;
+        }
+
+        private BoundBlock LowerBoundMultipleLocalDeclarationUsingVar(BoundMultipleLocalDeclarations boundMultiple,
+                                                                      ImmutableArray<LocalSymbol> locals,
+                                                                      List<BoundStatement> precedingStatements,
+                                                                      ImmutableArray<BoundStatement> followingStatements)
+        {
+            List<BoundLocalDeclaration> reversedLocals = Enumerable.Reverse(boundMultiple.LocalDeclarations).ToList();
```

Comment:
Both Lists should be `ArrayBuilder` and there's a `Reverse` method on `ArrayBuilder`